frontend/src/components/SepForm.js

```diff
@@ -106,15 +114,32 @@ const SepForm = observer(() => {
           <Form>
             <Grid container>
               <Grid item xs={12}>
-                <FormControl className={classes.siteId}>
+                <FormControl
+                  className={classes.siteId}
+                  error={errors.site_id && touched.site_id}
+                  disabled={participantId ? true : false}
+                >
                   <InputLabel htmlFor="site_id">Site ID</InputLabel>
-                  <Select id="site_id" name="site_id" value={values.site_id}>
-                    {participantStore.sites.map((siteId, i) => (
-                      <MenuItem key={i} value={siteId}>
-                        {siteId}
+                  <Select
+                    id="site_id"
+                    name="site_id"
+                    onChange={e => {
+                      handleChange(e)
+                      participantStore.setCurrentSite(e.target.value)
+                    }}
+                    value={values.site_id}
+                  >
+                    {sites.map((site, i) => (
+                      <MenuItem key={i} value={site.id}>
+                        {site.site_name}
```

Comment:
I'm currently getting a MUI error because the site_id value is available before the list of sites has been populated. "react_devtools_backend.js:2273 Material-UI: You have provided an out-of-range value `3` for the select (name="site_id") component."

Reply:
dont worry about this for now.